networkx/generators/random_graphs.py

```diff
@@ -962,3 +963,79 @@ def random_powerlaw_tree_sequence(n, gamma=3, seed=None, tries=100):
           "Exceeded max (%d) attempts for a valid tree sequence."%tries)
     return False
 
+
+def random_kernel_graph(n, kernel_integral, kernel_root=None, seed=None):
+    """Return an random graph based on the specified kernel.
+
+    The algorithm chooses each of the `[n(n-1)]/2` possible edges with
+    probability specified by a kernel `\kappa(x,y)` [1]_.  The kernel 
+    `\kappa(x,y)` must be a symmetric (in x,y), non-negative, bounded function.
+
+    Parameters
+    ----------
+    n : int
+        The number of nodes
+    kernal_integral : function
+        Function that returns the definite integral of the kernel `\kappa(x,y)`,
+        `F(y,a,b) := \int_a^b \kappa(x,y)dx`
+    kernel_root: function (optional)
+        Function that returns the root `b` of the equation `F(y,a,b) = r`.
+        If None, the root is found using scipy.optimize.brentq (requires scipy).
+    seed : int, optional
+        Seed for random number generator (default=None)
+
+    Notes
+    -----
+    The kernel is specified through its definite integral which must be 
+    provided as one of the arguments. If the integral and root of the 
+    kernal integral can be found in O(1) time then this algorithm runs in 
```

Comment:
Change to *O*(1) for proper italicization of the big O mathematical function. Same on the next line.